unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/BoxShape/ECSBoxShapeComponentHandler.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using DCL;
+using DCL.Controllers;
+using DCL.ECSRuntime;
+using DCL.Helpers;
+using DCL.Models;
+using UnityEngine;
+
+public class ECSBoxShapeComponentHandler : IECSComponentHandler<ECSBoxShape>
+{
+    private AssetPromise_PrimitiveMesh primitiveMeshPromisePrimitive;
+    internal MeshesInfo meshesInfo;
+
+    private bool isDisposed = false;
+    
+    public void OnComponentCreated(IParcelScene scene, IDCLEntity entity) { }
+    
+    public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
+    {
+        Dispose(scene);
+    }
+
+    public void Dispose(IParcelScene scene)
+    {
+        if (isDisposed)
+            return;
+        
+        isDisposed = true;
+        
+        if (primitiveMeshPromisePrimitive != null)
+            AssetPromiseKeeper_PrimitiveMesh.i.Forget(primitiveMeshPromisePrimitive);
+        
+        Utils.CleanMaterials(meshesInfo.meshRootGameObject.GetComponent<Renderer>());
+        meshesInfo.CleanReferences();
+        ECSComponentsUtils.RemoveRendereableFromDataStore(scene.sceneData.id,meshesInfo.rendereable);
+        meshesInfo = null;
+    }
+    
+    public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, ECSBoxShape model)
+    {
+        Mesh generatedMesh = null;
+        if (primitiveMeshPromisePrimitive != null)
+            AssetPromiseKeeper_PrimitiveMesh.i.Forget(primitiveMeshPromisePrimitive);
+
+        PrimitiveMeshModel primitiveMeshModelModel = new PrimitiveMeshModel(PrimitiveMeshModel.Type.Box);
```

Comment:
I don't like to have static variables for something that must be encapsulated, it could lead to bugs ( like the one that we have right now with the cube in the old ECS) and I don't think the small allocation is such a problem to justify the loss of encapsulation

Anyway, let's wait to @AjimenezDCL  to see what it thinks

Reply:
OK, disregard my comment, I just see that me need a new instance of the model for the asset promise, we can see to optimize that later